jupyter/nmm-tournament.ipynb

```diff
@@ -124,63 +194,86 @@
+   "metadata": {},
+   "source": [
+    "Die Hilfsfunktion `check_remis` überprüft ob der aktuelle Zustand und Spieler zu einem Remis führen. Ist dies der Fall wird ein entsprechendes `MatchResult` mit Begründung ansonsten `None` zurück gegeben. Gründe für ein Remis sind:\n",
+    "* die maximale Anzahl an Zügen wurde gespielt - `max_turns`;\n",
+    "* ein Zug wurde öfters als maximal erlaubt gespielt - `max_state_replayed`;\n",
+    "* die maximale Anzahl an Zügen ohne eine neue Mühle wurde gespielt - `max_states_without_mill`."
+   ]
+  },
   {
    "cell_type": "code",
    "execution_count": null,
    "id": "e68d843a",
    "metadata": {},
    "outputs": [],
    "source": [
-    "def check_remis(self: Round) -> Optional[RoundResult]:\n",
+    "def check_remis(self: Match) -> Optional[MatchResult]:\n",
     "    if len(self.log) >= self.max_turns:\n",
-    "        return RoundResult(\n",
+    "        return MatchResult(\n",
     "            winner = ' ',\n",
     "            log    = self.log,\n",
     "            reason = f\"Reached max_turns after {self.max_turns} turns\"\n",
     "        )\n",
     "    \n",
-    "    if self.visited[self.state] >= self.max_state_replayed:\n",
-    "        return RoundResult(\n",
+    "    if self.log.count(self.state) >= self.max_state_replayed:\n",
+    "        return MatchResult(\n",
     "            winner = ' ',\n",
     "            log    = self.log,\n",
-    "            reason = f\"State has been replayed for {self.visited[self.state]} turns\"\n",
+    "            reason = f\"State has been replayed for {self.log.count(self.state)} turns\"\n",
     "        )\n",
     "    \n",
     "    if self.no_mill_played >= self.max_states_without_mill:\n",
-    "        return RoundResult(\n",
+    "        return MatchResult(\n",
     "            winner = ' ',\n",
     "            log    = self.log,\n",
     "            reason = f\"No mill has been played for {self.no_mill_played} turns\"\n",
     "        )\n",
     "    return None\n",
     "\n",
-    "Round.check_remis = check_remis\n",
+    "Match.check_remis = check_remis\n",
     "del check_remis"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "id": "3372e05d",
+   "metadata": {},
+   "source": [
+    "Um ein Spiel zu spielen wird die Funktion `player` implementiert, diese bedient sich der im `Match` gespeicherten Einstellungen und Daten.\n",
+    "\n",
+    "Prinzipiell unendlich lang sind die Spieler abwechselnd am Zug und spielen ihren für am besten empfundenen Zug. Ein Spiel endet sobald ein per `finished` Funktion ein Gewinner ermittelt wurde oder per `check_remis` Hilfsfunktion ein Remis beschlossen wurde. Ist dies nicht der Fall wird der aktuelle Teilnehmer nach dem nächsten (besten) Zug befragt, dieser wird gespeichert und der Gegner ist am Zug.\n",
+    "\n",
+    "Da nach einer undefinierten Anzahl an Zügen entweder `finished` oder `check_remis` ein Ergebnis liefern, wird immer eine `MatchResult` Instanz zurück gegeben."
```

Comment:
```suggestion
    "Da nach einer undefinierten Anzahl an Zügen entweder `finished` oder `check_remis` ein Ergebnis liefern, wird immer eine `MatchResult` Instanz zurückgegeben."
```

Wenn bei dir irgendwas in deiner Schulzeit schiefgelaufen ist, dann dass du solche Wörter trennst xD

Reply:
Uff, ja das Bildungsministerium hat fragwürdige entscheidungen getroffen. Wir haben in NRW Rechtschreibung gelernt, indem wir erstmal nur so geschrieben haben, wie man spricht. Das ist vielleicht am Anfang einfacher, aber man bekommt sowas nicht mehr so einfach raus. Ich habe mal ein Grundschulheft von mir gefunden, wo ich `Froinde` geschrieben habe xD

Reply:
:joy: In Brandenburg wurde das zum Glück erst nach meiner Zeit eingeführt

Reply:
RIchtiger Mist.. Meine Eltern haben sich auch schon immer gewundert und habe trotzdem manchmal was gesagt, wenn etwas arg falsch war.